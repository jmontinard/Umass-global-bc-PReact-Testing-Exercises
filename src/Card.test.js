import { render } from "@testing-library/react";
import Card from './Card';

// Smoke Test
test('renders Card component without crashing', () => {
  render(<Card caption="Test Caption" src="test.jpg" currNum={1} totalNum={3} />);
});

// Snapshot Test
test('Card component snapshot', () => {
  const { asFragment } = render(<Card caption="Test Caption" src="test.jpg" currNum={1} totalNum={3} />);
  expect(asFragment()).toMatchSnapshot();
});


// Additional Test
test('renders Card component with correct content', () => {
  const { getByText, getByAltText } = render(
    <Card caption="Test Caption" src="test.jpg" currNum={1} totalNum={3} />
  );

  // ... other assertions ...

  // Check if the image is rendered with the correct alt text
  const imageElement = getByAltText('Test Caption');
  expect(imageElement).toBeInTheDocument();

  // Check if the image source contains the expected filename
  expect(imageElement.src).toContain('test.jpg');

  // ... other assertions ...
});

