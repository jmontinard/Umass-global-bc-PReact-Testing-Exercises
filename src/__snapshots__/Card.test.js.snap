// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Test Caption
    </h4>
    <img
      alt="Test Caption"
      class="Card-image"
      src="test.jpg"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;
